### PR TITLE
Add 'Subscription.test_iam_permissions' API wrapper.

### LINKED ISSUE #1073: Start implementing setIamPolicy, getIamPolicy, and testIamPermissions
I thought we already had an issue filed, must have mismembered.

See GoogleCloudPlatform/gcloud-node#758 and GoogleCloudPlatform/gcloud-node#767

@tseaver Care to take a stab at it?

### PR DESCRIPTION
**Pull request by tseaver**
~~Uses #1645 as a base.~~

Closes #1073.

### CODE REVIEW

**Comment by tseaver** [comment]
@dhermes Rebased after merging #1645 to `master`.  PTAL

**Comment by tseaver** [comment]
#1648 would cover the `test_iam_permissions` method name here, too.

File: docs/pubsub-usage.rst [hunk 1]
```diff
@@ -341,3 +341,15 @@ Update the IAM policy for a subscription:
...
+Test permissions allowed by the current IAM policy on a subscription:
+
+.. doctest::
+
+   >>> from gcloud import pubsub
+   >>> client = pubsub.Client()
+   >>> topic = client.topic('topic_name')
+   >>> subscription = topic.subscription('subscription_name')
+   >>> subscription.test_iam_permissions(
+   ...     ['roles/reader', 'roles/writer', 'roles/owner'])  # API request
```
Comment (dhermes) [inline docs/pubsub-usage.rst]: Sorry for the late question here. Would it be worth it to expose these constants publicly?

Comment (tseaver) [inline docs/pubsub-usage.rst]: 53f5ebf

Comment (dhermes) [inline docs/pubsub-usage.rst]: You should then use the public names in this doc?

Comment (tseaver) [inline docs/pubsub-usage.rst]: 537424e

Comment (tseaver) [inline docs/pubsub-usage.rst]: Hmm, having done 53f5ebf and 537424e, I'm not positive that exposing the constants is a win.

File: docs/pubsub-usage.rst [hunk 2]
```diff
@@ -341,3 +343,17 @@ Update the IAM policy for a subscription:
...
+.. doctest::
+
+   >>> from gcloud import pubsub
+   >>> from gcloud.pubsub.iam import OWNER_ROLE, WRITER_ROLE, READER_ROLE
+   >>> client = pubsub.Client()
+   >>> topic = client.topic('topic_name')
+   >>> subscription = topic.subscription('subscription_name')
+   >>> allowed = subscription.test_iam_permissions(
+   ...     [READER_ROLE, WRITER_ROLE, OWNER_ROLE])  # API request
+   >>> allowed == [READER_ROLE, WRITER_ROLE]
```
Comment (dhermes) [inline docs/pubsub-usage.rst]: @tseaver  RE: Constants not a win.

Why's that? You could at least make the import less heavy by using `iam.FOO_ROLE` in places.

Comment (tseaver) [inline docs/pubsub-usage.rst]: It isn't the "heaviness":  it just feels "wrong" to be exposing them, given that the `Policy` object allows manipulation without it.  Maybe I'm just not feelin' the usecase for `test_iam_permissions`.

Comment (dhermes) [inline docs/pubsub-usage.rst]: Yeah it's not very EAFP

**Comment by tseaver** [comment]
@dhermes any further issues?

**Comment by dhermes** [comment]
LGTM

**Pull request merged**